transit/script/bart.py

```diff
@@ -0,0 +1,396 @@
+
+    # Merge with other dataframes
+    BART_A = pd.merge(BART_A, nodes, on=["Node"], how="left")
+    BART_A = pd.merge(BART_A, station, on=["Node"], how="right")
+    BART = pd.merge(BART_A, BART_B, on=["Node", "TOD"], how="right")
+
+    # Drop rows with specific values
+    values_to_drop = ["Hillcrest eBART", "Coliseium OAC", "Somersville Road eBART"]
+    BART = BART[~BART["Node Name"].isin(values_to_drop)]
+
+    # Add columns and rearrange columns
+    BART["Key"] = BART["Station"] + BART["TOD"]
+    BART.columns = [
+        "Node",
+        "TOD",
+        "Boardings",
+        "Node Name",
+        "Station",
+        "Alightings",
+        "Key",
+    ]
+    BART = BART[
+        ["Node", "Node Name", "Station", "TOD", "Key", "Boardings", "Alightings"]
+    ]
+
+    # Sort and reset index
+    BART = BART[["Station", "TOD", "Key", "Boardings", "Alightings"]]
+    BART = BART.sort_values(by="Key").reset_index(drop=True)
+    BART.to_csv(output_dir / model_BART, index=False)
+    return BART
+
+
+# Function to map 'Station' to 'County'
+def map_station_to_county(station):
+    counties = {
+        "San Francisco": [
+            "EMBR",
+            "CIVC",
+            "24TH",
+            "MONT",
+            "POWL",
+            "GLEN",
+            "16TH",
+            "BALB",
+        ],
+        "San Mateo": ["DALY", "COLM", "SSAN", "SBRN", "SFIA", "MLBR"],
+        "Contra Costa": [
+            "RICH",
+            "ORIN",
+            "LAFY",
+            "WCRK",
+            "CONC",
+            "NCON",
+            "PITT",
+            "ANTC",
+            "DELN",
+            "PHIL",
+            "PCTR",
+            "PLZA",
+        ],
+        "Alameda": [
+            "WOAK",
+            "12TH",
+            "19TH",
+            "MCAR",
+            "ASHB",
+            "DUBL",
+            "WDUB",
+            "CAST",
+            "WARM",
+            "UCTY",
+            "SHAY",
+            "HAYW",
+            "BAYF",
+            "SANL",
+            "OAKL",
+            "COLS",
+            "FTVL",
+            "LAKE",
+            "ROCK",
+            "DBRK",
+            "NBRK",
+            "FRMT",
+        ],
+        "Santa Clara": [],  # Add stations for Santa Clara if available
+    }
+    for county, stations in counties.items():
+        if station in stations:
+            return county
+    return None
+
+
+def process_BART_county(
+    file_name, model_run_dir, output_dir, model_BART_county, model_BART
+):
+    BART_county = process_BART_data(file_name, model_run_dir, output_dir, model_BART)
+
+    # Add the 'County' column to the DataFrame
+    BART_county["County"] = BART_county["Station"].apply(
+        lambda x: map_station_to_county(x)
+    )
+    BART_county = (
+        BART_county.groupby(["County", "TOD"])[["Boardings", "Alightings"]]
+        .sum()
+        .reset_index()
+    )
+    BART_county.to_csv(output_dir / model_BART_county, index=False)
+
+
+# BART Screenline
```

Comment:
function docstring should be within function inside """ """

though, I think just saying `screenline` directly rather than `SL` would make the code clearer for someone just coming in and looking at the code (both here and elsewhere)

Reply:
priority is 👀 for first comment, can save for later for second comment